src/sentry/integrations/slack/notifications.py

```diff
@@ -154,52 +207,22 @@ def send_notification_as_slack(
         # TODO(mgaeta): We cannot determine the Slack workspace without an organization.
         return
 
-    data = get_channel_and_token_by_recipient(notification.organization, recipients)
-
-    for recipient, tokens_by_channel in data.items():
-        is_multiple = True if len([token for token in tokens_by_channel]) > 1 else False
-        if is_multiple:
-            logger.info(
-                "notification.multiple.slack_post",
-                extra={
-                    "notification": notification,
-                    "recipient": recipient.id,
-                },
-            )
-        extra_context = (extra_context_by_actor_id or {}).get(recipient.actor_id, {})
-        context = get_context(notification, recipient, shared_context, extra_context)
-        attachments = get_attachments(notification, recipient, context)
-
+    for recipient, tokens_by_channel in get_channel_and_token_by_recipient(
+        organization=notification.organization,
+        recipients=recipients,
+    ).items():
+        is_multiple = check_if_is_multiple(notification, recipient, tokens_by_channel)
+        attachments = get_attachments(
```

Comment:
@mgaeta wondering why this change is part of the PR, doesn't seem related to the description unless I'm missing something

Reply:
I agree, is it necessary to rewrite so much of this file?